src/debugAdapter/goDebug.ts

```diff
@@ -419,6 +419,17 @@ class Delve {
 					// Not applicable to exec mode in which case `program` need not point to source code under GOPATH
 					env['GOPATH'] = getInferredGopath(dirname) || env['GOPATH'];
 				}
+				if (
+					(env['GOOS'] && env['GOOS'] !== process.platform) ||
+					(env['GOARCH'] && env['GOARCH'] !== process.arch)
```

Comment:
It's not always true that Go's GOARCH and GOOS match process.arch/platform. 
For example, on my mac GOARCH=amd64 while process.arch=x64. So, need translation. 

BTW, delve doesn't return a meaningful error message in this case? I think we should file a bug upstream.

Reply:
What do you suggest doing here like keeping a list of compatible GOOS' and GOARCHs to make the comparison?